src/ADAL.PCL/AcquireTokenHandlerBase.cs

```diff
@@ -47,7 +47,8 @@ internal abstract class AcquireTokenHandlerBase
         protected AcquireTokenHandlerBase(RequestData requestData)
         {
             this.Authenticator = requestData.Authenticator;
-            this.CallState = CreateCallState(this.Authenticator.CorrelationId);
+            this.CallState = CreateCallState(this.Authenticator.CorrelationId, requestData.RequestId);
+            requestData.CorrelationId = this.CallState.CorrelationId;
```

Comment:
I am consuming it in the ApiEventHelper in AuthenticationContext class.I need it to track CorrelationId for telemetry.